plugins/follow_me/follow_me_impl.cpp

```diff
@@ -0,0 +1,294 @@
+    }
+}
+
+void FollowMeImpl::receive_param_follow_target_dist(bool success, float ft_dist_m)
+{
+    if (success) {
+        LogInfo() << "NAV_FT_DST: " << ft_dist_m << "m";
+        _config.set_follow_target_dist_m(ft_dist_m);
+    }
+}
+
+void FollowMeImpl::receive_param_follow_dir(bool success, int32_t dir)
+{
+    if (success) {
+        FollowMe::Configuration::FollowDirection d = FollowMe::Configuration::FollowDirection::NONE;
+        switch (dir) {
+            case 0: d = FollowMe::Configuration::FollowDirection::FRONT_RIGHT; break;
+            case 1: d = FollowMe::Configuration::FollowDirection::BEHIND; break;
+            case 2: d = FollowMe::Configuration::FollowDirection::FRONT; break;
+            case 3: d = FollowMe::Configuration::FollowDirection::FRONT_LEFT; break;
+            default: break;
+        }
+        _config.set_follow_dir(d);
+        LogInfo() << "NAV_FT_FS: " << FollowMe::Configuration::to_str(d);
+        if (d != FollowMe::Configuration::FollowDirection::NONE) {
+            _config.set_follow_dir(d);
+        }
+    }
+}
+
+void FollowMeImpl::receive_param_responsiveness(bool success, float rsp)
+{
+    if (success) {
+        LogInfo() << "NAV_FT_RS: " << rsp;
+        _config.set_responsiveness(rsp);
+    }
+}
+
+void FollowMeImpl::set_config(const FollowMe::Configuration &cfg)
+{
+    auto height = cfg.min_height_m();
+    auto dist = cfg.follow_target_dist_m();
+    int32_t dir = static_cast<int32_t>(cfg.follow_dir());
+    auto rsp = cfg.responsiveness();
+
+    LogInfo() << "Going to set min height " << height << " mts";
+    if (cfg.min_height_m() != FollowMe::Configuration::DEF_DIST_WRT_FT)
+        _parent->set_param_float_async("NAV_MIN_FT_HT", height,
+                                       std::bind(&FollowMeImpl::receive_param_min_height,
+                                                 this, _1, height));
+
+    if (cfg.follow_target_dist_m() != FollowMe::Configuration::DEF_DIST_WRT_FT)
+        _parent->set_param_float_async("NAV_FT_DST", dist,
+                                       std::bind(&FollowMeImpl::receive_param_follow_target_dist,
+                                                 this, _1, dist));
+    if (cfg.follow_dir() != FollowMe::Configuration::DEF_FOLLOW_DIR)
+        _parent->set_param_int_async("NAV_FT_FS", dir,
+                                     std::bind(&FollowMeImpl::receive_param_follow_dir,
+                                               this, _1, dir));
+    if (cfg.responsiveness() != FollowMe::Configuration::DEF_DYN_FLT_ALG_RSP)
+        _parent->set_param_float_async("NAV_FT_RS", rsp,
+                                       std::bind(&FollowMeImpl::receive_param_responsiveness,
+                                                 this, _1, rsp));
+}
+
+/**
+ * @brief FollowMeImpl::start
+ * @return
+ */
+FollowMe::Result FollowMeImpl::start()
+{
+    _parent->add_call_every(std::bind(&FollowMeImpl::timeout_occurred, this), 0.6f, &_timeout_cookie);
```

Comment:
I see, this is a call every. Then I would rename the `timeout_occured` method.

Reply:
ok